packages/embark-compiler/src/index.ts

```diff
@@ -14,75 +16,166 @@ class Compiler {
-            someCb(null, true);
-          });
-        }, (err: Error, result: boolean) => {
-          if (err) {
-            return next(err);
-          }
-          if (!result) {
-            // No compiler was compatible
-            return next(new Error(__("No installed compiler was compatible with your version of %s files", extension)));
-          }
-          next();
-        });
-      },
-      (err: any) => {
-        contractFiles.filter((f: any) => !f.compiled).forEach((file: any) => {
-          this.logger.warn(__("%s doesn't have a compatible contract compiler. Maybe a plugin exists for it.", file.path));
-        });
+              async.someLimit(
+                compilers,
+                1,
+                (compiler: any, someCb: Callback<boolean>) => {
+                  compiler.call(
+                    compiler,
+                    matchingFiles,
+                    compilerOptions,
+                    (err: any, compileResult: any) => {
+                      if (err) {
+                        return someCb(err);
+                      }
+                      if (compileResult === false) {
+                        // Compiler not compatible, trying the next one
+                        return someCb(null, false);
+                      }
+                      Object.assign(compiledObject, compileResult);
+                      someCb(null, true);
+                    },
+                  );
+                },
+                (err: Error, result: boolean) => {
+                  if (err) {
+                    return nextObj(err);
+                  }
+                  if (!result) {
+                    // No compiler was compatible
+                    return nextObj(
+                      new Error(
+                        __(
+                          "No installed compiler was compatible with your version of %s files",
+                          extension,
+                        ),
+                      ),
+                    );
+                  }
+                  nextObj();
+                },
+              );
+            },
+            (err?: Error | null) => {
+              if (err) {
+                return next(err);
+              }
 
-        cb(err, compiledObject);
+              files
+                .filter((f: any) => !f.compiled)
+                .forEach((file: any) => {
+                  this.logger.warn(
+                    __(
+                      "%s doesn't have a compatible contract compiler. Maybe a plugin exists for it.",
+                      file.path,
+                    ),
+                  );
+                });
+              next(null, compiledObject);
+            },
+          );
+        },
+        (compiledObject: any, next: Callback<any>) => {
+          this.plugins.runActionsForEvent(
+            "compiler:contracts:compile:after",
+            compiledObject,
+            (err?: Error | null, compiledObj: any = {}) => {
```

Comment:
I think you can just pass `next` straight up
```
 this.plugins.runActionsForEvent "compiler:contracts:compile:after",compiledObject, next);
```

Reply:
Ah, great point, let me check. I had kept them more explicit because it took a little while to get things working and the callbacks were good places to put breakpoints and/or use logger, but now it should be cleaned up where possible.

Reply:
Done.